components/script/dom/htmlinputelement.rs

```diff
@@ -1193,7 +1193,47 @@ impl HTMLInputElement {
             },
             // https://html.spec.whatwg.org/multipage/#range-state-(type=range):value-sanitization-algorithm
             InputType::Range => {
-                value.set_best_representation_of_the_floating_point_number();
+                let minimum = self.Min().parse().unwrap_or(0f64);
+                let maximum = self.Max().parse().unwrap_or(100f64);
+
+                let new_v = if let Ok(value) = value.trim().parse::<f64>() {
+                    if value < minimum || maximum < minimum {
+                        minimum
+                    } else if value > maximum {
+                        maximum
+                    } else {
+                        value
+                    }
```

Comment:
I think this can be a shorter expression:

```rust
value.min(maximum).max(minimum)
```

Clamping by the maximum before the minimum covers the case where the former is less than the latter.